File: cpp/05_permutations.cpp
#include <iostream>

using namespace std;

int main() {
    int n;
    cin >> n;

    if (n == 1) {
        cout << 1;
        return 0;
    }

    if (n <= 3) {
        cout << "NO SOLUTION";
        return 0;
    }

    int value = 2;
    while (value <= n) {
        cout << value << ' ';
        value += 2;
    }

    value = 1;
    while (value <= n) {
        cout << value << ' ';
        value += 2;
    }

    return 0;
}

File: rust/src/bin/05_permutations.rs
use std::io;

fn main() {
    let mut line = String::new();
    let _ = io::stdin().read_line(&mut line);

    let n: u32 = line.trim().parse().unwrap();

    if n == 1 {
        print!("{}", 1);
        return;
    }

    if n <= 3 {
        print!("NO SOLUTION");
        return;
    }

    (2..=n)
        .filter(|x| x % 2 == 0)
        .for_each(|x| print!("{} ", x));

    (1..=n)
        .filter(|x| x % 2 != 0)
        .for_each(|x| print!("{} ", x));
}
